netket/stats/mc_stats.py

```diff
@@ -152,48 +152,53 @@ def _get_blocks(data, block_size):
     return data[:, 0 : n_blocks * block_size].reshape((-1, block_size)).mean(axis=1)
 
 
-def _block_variance(data, l):
+def _block_variance(data, l, *, token=None):
     blocks = _get_blocks(data, l)
     ts = _total_size(blocks)
     if ts > 0:
-        return _var(blocks), ts
+        res, token = _var(blocks, token=token)
+        return res, ts, token
     else:
-        return jnp.nan, 0
+        return jnp.nan, 0, token
 
 
-def _batch_variance(data):
+def _batch_variance(data, *, token=None):
     b_means = data.mean(axis=1)
     ts = _total_size(b_means)
-    return _var(b_means), ts
+    res, token = _var(b_means, token=token)
+    return res, ts, token
 
 
-def _split_R_hat(data, W):
+def _split_R_hat(data, W, *, token=None):
     N = data.shape[-1]
     if not config.netket_use_plain_rhat:
         # compute split-chain batch variance
         local_batch_size = data.shape[0]
         if N % 2 == 0:
             # split each chain in the middle,
             # like [[1 2 3 4]] -> [[1 2][3 4]]
-            batch_var, _ = _batch_variance(data.reshape(2 * local_batch_size, N // 2))
+            batch_var, _, token = _batch_variance(
+                data.reshape(2 * local_batch_size, N // 2), token=token
+            )
         else:
             # drop the last sample of each chain for an even split,
             # like [[1 2 3 4 5]] -> [[1 2][3 4]]
-            batch_var, _ = _batch_variance(
-                data[:, :-1].reshape(2 * local_batch_size, N // 2)
+            batch_var, _, token = _batch_variance(
+                data[:, :-1].reshape(2 * local_batch_size, N // 2),
+                token=token,
             )
 
     # V_loc = _np.var(data, axis=-1, ddof=0)
     # W_loc = _np.mean(V_loc)
     # W = _mean(W_loc)
     # # This approximation seems to hold well enough for larger n_samples
-    return jnp.sqrt((N - 1) / N + batch_var / W)
+    return jnp.sqrt((N - 1) / N + batch_var / W), token
 
 
 BLOCK_SIZE = 32
 
 
-def statistics(data):
+def statistics(data, *, token=None, return_token=False):
```

Comment:
this could be simplified down to `token=None/token/True`. token=`True` would be equivalent to `return_token=True` with `token=None`...